src/QueueProvider.php

```diff
@@ -0,0 +1,132 @@
+<?php
+declare(strict_types=1);
+
+namespace Yiisoft\Queue\Redis;
+
+use RedisException;
+use Yiisoft\Queue\Redis\Exception\NotConnectedRedisException;
+
+class QueueProvider implements QueueProviderInterface
+{
+    private const DEFAULT_CHANNEL_NAME = 'yii-queue';
+
+    /**
+     * @throws RedisException
+     */
+    public function __construct(
+        private \Redis $redis, //redis connection,
+        private string $channelName = self::DEFAULT_CHANNEL_NAME
+    )
+    {
+        if (!$this->redis->isConnected()) {
```

Comment:
Should not try to connect in constructor or without an intent to ask data

Reply:
Did I understand correctly: it is necessary to check the isConnected immediately before accessing the radish?
For example:
```php
public function getId(): int
    {
        if(!$this->redis->isConnected()){
            throw new NotConnectedRedisException('Redis is not connected');
        }
        $id = $this->redis->incr("$this->channelName.message_id");
        if (is_int($id)) {
            return $id;
        }

        throw new \RuntimeException('Unable to get message id');
    }
```
And so in every function?

Reply:
Implemented checking in all public methods that use redis client.